tcms/testcases/tests/test_views.py

```diff
@@ -564,3 +566,33 @@ def test_disabled_priority_now_shown(self):
         self.assertEqual(HTTPStatus.OK, response.status_code)
         self.assertContains(response, 'Set P3')
         self.assertNotContains(response, 'Set P4')
+
+
+class TestShowCases(BasePlanCase):
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        cls.request = RequestFactory()
+
+    def test_get_selected_testcases_works_with_string_pks(self):
+        """
+        Assures that tcms.testcases.views.get_selected_testcases
+        returns the same results, regardless of whether the
+        passed request contains the case pks as strings or
+        integers, as long as they are the same pks.
+        """
+
+        case_int_pks = [1, 2, 3, 4, 5, 6]
+        case_str_pks = []
+
+        for _pk in case_int_pks:
+            case_str_pks.append(str(_pk))
+
+        int_pk_query = self._get_selected_testcases(case_int_pks)
+        str_pk_query = self._get_selected_testcases(case_str_pks)
+
+        for case in TestCase.objects.filter(pk__in=case_int_pks):
+            self.assertTrue(case in int_pk_query and case in str_pk_query)
```

Comment:
Use two assert statements, one for each condition. When things fail the traceback will point to the specific assert that failed and you will immediately know which one of the conditions failed. This will be easier to debug,